### PR TITLE
updates

### PR DESCRIPTION
**Pull request by turuslan**
### Description of the Change
- update blocksync
- update market and power actor states
- add const file
- update weight and fee function
- update some api
- add pubsub and empty main

### CODE REVIEW

File: core/markets/storage/client/impl/storage_market_client_impl.cpp [hunk 1]
```diff
@@ -109,7 +109,7 @@ namespace fc::markets::storage::client {
     std::vector<StorageProviderInfo> storage_providers;
     for (const auto &miner_address : miners) {
       OUTCOME_TRY(miner_info, api_->StateMinerInfo(miner_address, tipset_key));
-      PeerInfo peer_info{.id = miner_info.peer_id, .addresses = {}};
+      PeerInfo peer_info{.id = *miner_info.peer_id, .addresses = {}};
```
Comment (artyom-yurin) [inline core/markets/storage/client/impl/storage_market_client_impl.cpp:112]: What if miner_info is none?

**Review by artyom-yurin — APPROVED** [review]

File: core/markets/storage/client/impl/storage_market_client_impl.cpp (hunk 1 above)
Comment (turuslan) [reply core/markets/storage/client/impl/storage_market_client_impl.cpp:112]: Don't know, lotus has no check there.

File: core/api/api.hpp [hunk 2]
```diff
@@ -270,6 +272,16 @@ namespace fc::api {
     uint64_t partition;
   };
 
+  struct MinerInfo2 {
```
[the remark this answers is not in the document]
Comment (Alexey-N-Chernyshov) [reply core/api/api.hpp:275]: +1

[the remark this answers is not in the document]
Comment (Alexey-N-Chernyshov) [reply core/api/api.hpp:275]: it is identical to fc::vm::actor::builtin::miner::MinerInfo

File: core/const.cpp [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "const.hpp"
+
+#define DEFINE(x) decltype(x) x
+
+namespace fc {
+  DEFINE(kConsensusMinerMinPower){BigInt{1} << 40};
```
Comment (Alexey-N-Chernyshov) [inline core/const.cpp:11]: need macro for this?

Comment (turuslan) [reply core/const.cpp:11]: With this macro you don't need to repeat type.

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**